lib/adapters/JasmineAdapter.js

```diff
@@ -4,68 +4,115 @@ import {Test, Suite} from "../Data.js";
 export default class JasmineAdapter extends EventEmitter {
     constructor(jasmine) {
         super();
+        this.jasmine = jasmine;
         jasmine.addReporter({
+            jasmineStarted: this.onJasmineStarted.bind(this),
             specDone: this.onSpecDone.bind(this),
             specStarted: this.onSpecStarted.bind(this),
             suiteStarted: this.onSuiteStarted.bind(this),
             suiteDone: this.onSuiteDone.bind(this),
             jasmineDone: this.onJasmineDone.bind(this)
         });
-        // a stack of test arrays. The top element on the stack is the currently active suite
-        this.tests = [[]];
-        // a stack of suite arrays
-        this.suites = [[]];
+
+        // Map internal jasmine ids to standardized test/suite objects.
+        this.idObjectMap = {};
     }
 
+    /**
+     * Recursively converts a jasmine suite into a standardized suite.
+     */
+    convertSuite(jasmineSuite) {
+        var name = jasmineSuite.description;
+        var childSuites = [];
+        var tests = [];
+        for (let child of jasmineSuite.children) {
+            // a child can be either a suite or a spec. We use the id to distinguish between the two.
+            if (child.id.indexOf("suite") === 0) {
+                childSuites.push(this.convertSuite(child));
+            } else if (child.id.indexOf("spec") === 0) {
+                tests.push(this.convertSpec(child));
+            }
```

Comment:
There is no else here. Can the children property contain any properties that are neither suite nor spec?